File: PoC_observer/GerenciadorNotificacoes.java
import java.util.ArrayList;

public class GerenciadorNotificacoes {
    private ArrayList<INotificavel> notificaveis = new ArrayList<INotificavel>();
    private Usuario usuario;

    public GerenciadorNotificacoes(Usuario usuario) {
        this.usuario = usuario;
    }

    public void adicinar(INotificavel notificavel) {
        notificaveis.add(notificavel);
    }

    public void notificarTodos(String msg) {
        for (INotificavel notificavel : notificaveis) {
            notificavel.notificar(msg, this.usuario);
        }
    }
}

File: teste/src/GerenciadorNotificacoes.cpp
#include <iostream>
#include "../include/GerenciadorNotificacoes.hpp"
#include "../include/Usuario.hpp"

// Construtor
GerenciadorNotificacoes::GerenciadorNotificacoes(Usuario* usuario) : usuario(usuario) {}

void GerenciadorNotificacoes::adicionar(Usuario* notificavel) {
    notificaveis.push_back(notificavel);
}

void GerenciadorNotificacoes::notificarTodos(std::string msg) {
    for (Usuario* notificavel : notificaveis) {
        notificavel->notificar(msg, usuario->getNome());
    }
}
